projects/plugins/social/src/js/components/admin-page/index.jsx

```diff
@@ -18,6 +18,59 @@ import React from 'react';
  */
 import styles from './styles.module.scss';
 
+const ConnectionItem = props => {
+	const { connections } = window.jetpackSocialInitialState;
+
+	return props.connectionIds.map( connectionId => {
+		const connection = connections[ props.provider ][ connectionId ];
+		return (
+			<div
+				key={ connectionId + Date.now().toString( 36 ) + Math.random().toString( 36 ).substr( 2 ) }
+			>
+				<tr>
+					<th className={ styles.connectionRow }> { __( 'Name', 'jetpack-social' ) }</th>
+					<th className={ styles.connectionRow }>{ __( 'Image', 'jetpack-social' ) }</th>
+				</tr>
+				<tr>
+					<td className={ styles.connectionRow }>{ connection.external_display }</td>
+					<td className={ styles.connectionRow }>
+						{ connection.profile_picture && (
+							<img
+								alt="connection avatar"
+								src={ connection.profile_picture }
+								height="50px"
+								width="50px"
+							/>
+						) }
+					</td>
+				</tr>
+			</div>
+		);
+	} );
+};
+
+const ConnectionItems = () => {
+	const { connections } = window.jetpackSocialInitialState;
+
+	if ( ! connections ) {
+		return null;
+	}
+
+	const providers = Object.keys( connections );
+	return providers.map( provider => {
+		return (
+			<div key={ provider + Date.now().toString( 36 ) + Math.random().toString( 36 ).substr( 2 ) }>
```

Comment:
Similarly, I think just the provider name would be best here, maybe with something prepended, so it's unlikely to clash with something else on the page.

Reply:
This is done!